app/controllers/UpfrontPaymentController.scala

```diff
@@ -77,46 +81,77 @@ class UpfrontPaymentController @Inject() (
               routes.MonthlyPaymentAmountController.monthlyPaymentAmount()
             }
           journeyService.updateCanPayUpfront(request.journeyId, canPayUpfront)
-            .map(_ => Redirect(pageToRedirectTo))
+            .map(_ => Redirect(pageToRedirectTo.url))
         }
       )
   }
 
-  val upfrontPaymentAmount: Action[AnyContent] = as.authenticatedJourneyAction.async { implicit request =>
-    val mockJourney = MockJourney(userAnswers = UserAnswers.empty.copy(hasUpfrontPayment = Some(true)))
-    Future.successful(Ok(views.upfrontPaymentAmountPage(upfrontPaymentAmountForm(mockJourney), mockJourney.qualifyingDebt, AmountInPence(100L))))
+  val upfrontPaymentAmount: Action[AnyContent] = as.eligibleJourneyAction { implicit request =>
+    request.journey match {
+      case j: Journey.BeforeAnsweredCanPayUpfront                                      => logErrorAndRouteToDefaultPage(j)
+      case j: Journey.AfterAnsweredCanPayUpfront if !j.canPayUpfront.userCanPayUpfront => logErrorAndRouteToDefaultPage(j)
+      case j: Journey.AfterAnsweredCanPayUpfront if j.canPayUpfront.userCanPayUpfront  => displayUpfrontPageAmountPage(j)
+    }
+  }
+
+  private val minimumUpfrontPaymentAmount: AmountInPence = appConfig.JourneyVariables.minimumUpfrontPaymentAmountInPence
+
+  private def displayUpfrontPageAmountPage(journey: Journey)(implicit request: Request[_]): Result = {
+    val backUrl: Option[String] = Some(routes.UpfrontPaymentController.canYouMakeAnUpfrontPayment().url)
+    val debtTotalAmount: DebtTotalAmount = UpfrontPaymentController.determineMaxDebt(journey) match {
+      case Some(value) => value
+      case None        => Errors.throwBadRequestException("I am an error, this should never happen")
```

Comment:
`displayUpfrontPageAmountPage` is called at a point where a `Journey.AfterAnsweredCanPayUpfront` is available. If you make `UpfrontPaymentController.determineMaxDebt` take in that type instead does that get rid of one of the ways an error can occur?

In the error message is actually seen a more useful message would be better - something like `"Could not determine max debt"`?

Reply:
just implemented this, resolving once I've pushed